core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -573,7 +573,10 @@ def _preprocess_fit_resources(self, silent=False, total_resources=None, parallel
                 )
                 minimum_model_num_cpus = minimum_model_resources.get('num_cpus', 1)
                 enforced_num_cpus = max(minimum_model_num_cpus, enforced_num_cpus - 2)  # leave some cpu resources for process running by cluster nodes
-                kwargs["num_cpus"] = enforced_num_cpus
+            max_num_cpus = self._get_maximum_resources().get("num_cpus", None)
+            if max_num_cpus is not None:
+                enforced_num_cpus = min(max_num_cpus, enforced_num_cpus)
+            kwargs["num_cpus"] = enforced_num_cpus
```

Comment:
do this for gpus too (make sure only 1 call to the method though)

Reply:
This actually gives a very elegant way of disabling GPUs for models, we can simply set `ag.max_gpus=0` for models that don't have GPU support.